conn.go

```diff
@@ -1078,10 +1110,15 @@ func (c *Conn) exec(ctx context.Context, req frameBuilder, tracer Tracer) (*fram
 
 	err := req.buildFrame(framer, stream)
 	if err != nil {
+		// closeWithError will block waiting for this stream to either receive a response
+		// or for us to timeout.
+		close(call.timeout)
 		// We failed to serialize the frame into a buffer.
 		// This should not affect the connection as we didn't write anything. We just free the current call.
 		c.mu.Lock()
-		delete(c.calls, call.streamID)
+		if !c.closed {
```

Comment:
question: Why are we adding this condition?

Reply:
Commit https://github.com/kiwicom/gocql/pull/19/commits/46d432e97daf22e898f6d9f1044100d12eb2e039 makes closeWithError take ownership of `c.calls` and swaps it with `nil` map. While [delete on a nil-map is a no-op](https://golang.org/ref/spec#Deletion_of_map_elements), I thought it would be better to explicitly not touch `c.calls` after `closeWithError` steals the map (we need the check when adding elements, as assignment to nil map panics, this keeps the check consistently in all places touching `c.calls`).